fabric-content-registries-v0/src/main/java/net/fabricmc/fabric/api/registry/SpreadableBlockRegistry.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.registry;
+
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.fabric.api.util.Block2ObjectMap;
+import net.fabricmc.fabric.impl.content.registry.SpreadableBlockRegistryImpl;
+
+/**
+ * Registries of blocks to block states, defining the block state to replace a
+ * block with when a particular type of other block spreads to it.
+ *
+ * <p>For example, to get the registry for Mycelium block spread
+ * and then register a modded block pair:
+ *
+ * <pre>{@code
+ * SpreadableBlockRegistry.getOrCreateInstance(SpreadableBlockRegistry.MYCELIUM)
+ *        .add(MyModBlocks.MY_DIRT, MyModBlocks.MY_MYCELIUM.getDefaultState());
+ * }</pre>
+ */
+@ApiStatus.NonExtendable
+public interface SpreadableBlockRegistry extends Block2ObjectMap<BlockState> {
+	/**
+	 * Registry ID for Minecraft Grass Block type spreadable blocks.
+	 */
+	Identifier GRASS = new Identifier("grass");
+	/**
+	 * Registry ID for Minecraft Mycelium Block type spreadable blocks.
+	 */
+	Identifier MYCELIUM = new Identifier("mycelium");
+
+	/**
+	 * Get the registry for a given registry ID, or create a new registry for the ID
+	 * if none currently exists.
+	 *
+	 * @param type The registry type Identifier for the desired spreadable block registry
+	 * @return The SpreadableBlockRegistry for the given ID
+	 */
+	static SpreadableBlockRegistry getOrCreateInstance(Identifier type) {
```

Comment:
Maybe just `get` here? Im not sure the `OrCreateInstance` is really that useful to the API user.

Reply:
It creates so the API user can add a custom SpreadableBlock type.  While I don't have a specific need for this myself at the moment, I can imagine a lot of potential uses.  The test cases I provided include creating a spreadable version of Podzol that spreads to Coarse Dirt as an example.

Reply:
That name does seem too long considering there's no `get` that doesn't have the "or create" functionality.